tests/mocks/MockFileSystem.hh

```diff
@@ -53,6 +59,55 @@ public:
         static const std::string sEmpty;
         return sEmpty;
     }
+    
+    /// <summary>
+    /// Mocks the size of the file.
+    /// </summary>
+    /// <param name="sPath">The path of the file.</param>
+    /// <param name="nFileSize">The size of the file.</param>
+    /// <remarls>Mocked size will be discarded if the file is updated (MockFile, CreateTextFile, or AppendTextFile is called for the file).</remarks>
+    void MockFileSize(const std::wstring& sPath, int64_t nFileSize)
+    {
+        m_mFileSizes.insert_or_assign(sPath, nFileSize);
+    }
+
+    int64_t GetFileSize(const std::wstring& sPath) const override
+    {
+        auto pIter = m_mFileSizes.find(sPath);
+        if (pIter != m_mFileSizes.end())
+            return pIter->second;
+
+        auto pIter2 = m_mFileContents.find(sPath);
```

Comment:
It's up to you but `pIter `and `pIter2 `can be destroyed earlier if the conditions failed.